src/coordination/coordinator_instance.cpp

```diff
@@ -323,6 +323,15 @@ auto CoordinatorInstance::ForceResetCluster_() -> ForceResetClusterStateStatus {
 
   auto const instances = raft_state_->GetReplicationInstances();
 
+  // There is nothing to do if we don't have any instances
+  if (instances.empty()) {
+    if (!raft_state_->AppendCloseLock()) {
+      spdlog::error("Aborting force reset as we failed to close lock on action.");
+      return ForceResetClusterStateStatus::FAILED_TO_CLOSE_LOCK;
+    }
+    return ForceResetClusterStateStatus::SUCCESS;
+  }
+
```

Comment:
Can we move this at the function's start so we don't do any non-necessary work if there are no instances?

Reply:
Actually, it should be here as we need to check if we added instance to repl_instances_ and stop checks there, and also there is lock opening and other things needed which are before, so this is correct spot in my opinion